frontend/app/components/mapbox/carto-vector-source.js

```diff
@@ -0,0 +1,108 @@
+import Component from '@ember/component';
+import fetch from 'fetch';
+import { action, computed } from '@ember-decorators/object';
+
+export const CARTO_USERNAME = 'planninglabs';
+export const CARTO_MAPS_ENDPOINT = `https://${CARTO_USERNAME}.carto.com/api/v1/map`;
+
+/**
+ *
+ * MapboxCartoLayersComponent renders vector tile layers via e-mapbox-gl.
+ * It provides a contextual component necessary for setting up layers:
+ *
+ * ```
+ * {{#mapbox/carto-vector-source map=map as |carto-layer|}}
+ *   {{carto-layer
+ *     sql='select * from geodata'
+ *     layer=(hash fill-color='blue')
+ *   }}
+ * {{/mapbox/carto-vector-source}}
+ * ```
+ *
+ * Currently this can't handle subsequent invocations of carto-layers because
+ * of a limitation with the ember-mapbox-gl bindings lib and there isn't an
+ * API for it. See https://github.com/mapbox/mapbox-gl-js/pull/8048.
+ */
+export default class MapboxCartoLayersComponent extends Component {
+  // ember hook, triggered on render. used here to trigger template URL handshake.
+  didInsertElement() {
+    this.fetchTemplate();
+  }
+
+  // public
+  // required
+  // e-mapbox-gl map object
+  map = null;
+
+  // public
+  // options used for the mapbox-gl source definition
+  options = {};
+
+  // private
+  _registeredLayers = [];
+
+  // resolved tilejson url templates from carto
+  _tiles = [];
+
+  // sets up source according mapbox-gl source specification
+  @computed('_tiles')
+  get mapboxSourceOptions() {
+    return {
+      type: 'vector',
+      tiles: this._tiles,
+      ...this.options,
+    };
+  }
+
+  // options object used for Carto Maps API.
+  // creates an entry in the layers property for
+  // each child layer component that depends on child layers ids
+  @computed('_registeredLayers')
+  get cartoAnonMapOptions() {
```

Comment:
what about this map is Anon?

Reply:
Oooh it's just named from this: https://carto.com/developers/maps-api/reference/#tag/Anonymous-Maps

It's literally "Carto Anonymous Map Options"

Reply:
"Anonymous Maps" is what Carto calls maps generated by a SQL request, as opposed to maps with a saved, named query. https://carto.com/developers/maps-api/reference/#tag/Anonymous-Maps